Make regalloc visit fallthrough_return instructions.

sunfishcode opened the pull request:
Add an explicit "is_ghost" property to selected instructions, and use
that to determine whether reload and coloring should visit instructions.
This allows them to visit fallthrough_return instructions and insert
fills and register moves as needed.

bnjbvr commented on lib/codegen/meta-python/cdsl/instructions.py:
```diff
@@ -108,6 +109,7 @@ class Instruction(object):
             'True for all indirect branch or jump instructions.',
             'is_call': 'Is this a call instruction?',
             'is_return': 'Is this a return instruction?',
+            'is_ghost': 'Is this a ghost instruction?',
```
It'd be nice to precise what it means: my understanding is that it's an instruction with an illegal encoding that should not be visited by register allocation. (fallthrough_return has an illegal encoding but must be visited -- are there any other ones like fallthrough_return?)

bnjbvr commented:
Also, please update the commit message so it doesn't include "unreachable_return" anymore :). Thanks!

sunfishcode commented:
Yes, your understanding is exactly correct.

sunfishcode commented:
Ok, I've now updated the commit message, added a verifier check that ghost instructions don't have an encoding, and added better comments explaining what ghost instructions are.

bnjbvr commented:
Great, thanks!
